lib/iris/fileformats/cf.py

```diff
@@ -487,8 +487,8 @@ def identify(cls, variables, ignore=None, target=None, warn=True, monotonic=Fals
             # String variables can't be coordinates
             if _is_str_dtype(nc_var):
                 continue
-            # Restrict to one-dimensional with name as dimension OR zero-dimensional scalar
-            if not ((nc_var.ndim == 1 and nc_var_name in nc_var.dimensions) or (nc_var.ndim == 0)):
+            # Restrict to one-dimensional with name as dimension
```

Comment:
A zero dimensional scalar variable can never be a netcdf coordinate variable.  The definitions within the CF Conventions and the netCDF user guide are explicit on this.  
A 'coordinate variable' is one dimensional, with the same name as the dimension it is defining.
A scalar variable is not defined with respect to a dimension, it is a scalar.